### DISCUSSION TITLE
How to save the best model instead of the last?

### DISCUSSION THREAD

**Discussion by Caspeerrr** [facebookresearch/detectron2#3927] [Q&A]
I want to save the best model instead of the last model for detectron2. The evaluation metric I want to use is AP50 or something similar. The code I currently have is:
`trainer.register_hooks([
            EvalHook(eval_period=20, eval_function=lambda:{'AP50':function?}), 
            BestCheckpointer(eval_period=20, checkpointer=trainer.checkpointer, val_metric= "AP50", mode="max")
            ])
`
But I have no idea what I have to substitute for the function in EvalHook. I use a subset of the coco dataset to train the model, and I saw that detectron2 contains some evaluation measures for the coco dataset, but I have no idea how to implement this.

**Comment by zhanghang1989** [comment]
You may check the tensorboard history to find the best one

**Comment by Aayushjshah** [comment]
**did you find any fix Casperrrr?**

**Comment by aymanaboghonim** [comment]
did you find a proper way to do your job ???
